runtime/lib/runtime.cpp

```diff
@@ -6,26 +6,40 @@
 #include "tt/runtime/utils.h"
 #include "ttmlir/Version.h"
 
-#if defined(TT_RUNTIME_ENABLE_TTNN) && defined(TT_RUNTIME_ENABLE_TTMETAL)
-#error                                                                         \
-    "Only one of TT_RUNTIME_ENABLE_TTNN and TT_RUNTIME_ENABLE_TTMETAL can be defined"
-#endif
-
 #if defined(TT_RUNTIME_ENABLE_TTNN)
 #include "tt/runtime/detail/ttnn.h"
-#elif defined(TT_RUNTIME_ENABLE_TTMETAL)
+#endif
+
+#if defined(TT_RUNTIME_ENABLE_TTMETAL)
 #include "tt/runtime/detail/ttmetal.h"
 #endif
 
 namespace tt::runtime {
+
+void setCompatibleRuntime(const Binary &binary) {
+  std::string_view fileIdentifier = binary.getFileIdentifier();
+  if (fileIdentifier == "TTNN") {
```

Comment:
`if (fileIdentifier == ::tt::Target::ttnn::TTNNBinaryExtension())`

Reply:
Also I think we should wrap these if statements with `defined(TT_RUNTIME_ENABLE_TTNN)` guards so that you can't set a runtime that's not compiled in.

Reply:
Great point! Completely forgot about defined(TT_RUNTIME_ENABLE_TTNN)